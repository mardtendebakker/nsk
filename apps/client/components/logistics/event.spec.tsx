import { fireEvent, render } from '@testing-library/react';
import Event from './event';

function RenderComponent({ onClick }:{ onClick:()=> void }) {
  return (
    <Event
      type="delivery"
      logisticService={{
        event_date: '',
        event_title: 'eventTitle',
        id: 1,
        logistic: {
          id: 1,
          username: 'string',
          firstname: 'string',
          lastname: 'string',
          email: 'string',
        },
        order: {
          id: 1,
          order_nr: 'string',
          order_status: { color: 'color' },
          supplier: {
            city: 'string',
            country: 'string',
            state: 'string',
            zip: 'string',
            street: 'string',
            phone: 'string',
            name: 'string',
            email: 'string',
            representative: 'string',
          },
          customer: {
            city: 'string',
            country: 'string',
            state: 'string',
            zip: 'string',
            street: 'string',
            phone: 'string',
            name: 'string',
            email: 'string',
            representative: 'string',
          },
        },
      }}
      top="1rem"
      height="1rem"
      left="1rem"
      width="1rem"
      onClick={onClick}
    />
  );
}

describe('event', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<RenderComponent onClick={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('handles click', () => {
    const onClick = jest.fn();
    const { getByTestId } = render(<RenderComponent onClick={onClick} />);
    fireEvent.click(getByTestId('event'));
    expect(onClick).toHaveBeenCalled();
  });
});
