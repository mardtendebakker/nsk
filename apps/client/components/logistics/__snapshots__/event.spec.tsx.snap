// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`event matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-eu1845"
    data-testid="event"
  >
    <div
      class="MuiBox-root css-cvabll"
      data-mui-internal-clone-element="true"
    >
      <p
        class="MuiTypography-root MuiTypography-inherit css-37jdci-MuiTypography-root"
      >
        eventTitle, 
      </p>
    </div>
  </div>
</DocumentFragment>
`;
